File: RPiWindows/Controllers/MovementController.cs
using RPiWindows.LANClients;
using RPiWindows.Models;

namespace RPiWindows.Controllers
{
    class MovementController
    {
        // WASD, for forward, left, back, right. 
        // qezx for foward left, forward right, backward left, and backward right
        const string DRIVE_FORWARD = "W";
        const string DRIVE_FORWARD_LEFT = "q";
        const string DRIVE_FORWARD_RIGHT = "e";
        const string DRIVE_BACKWARD = "S";
        const string DRIVE_BACKWARD_LEFT = "z";
        const string DRIVE_BACKWARD_RIGHT = "x";
        const string TURN_LEFT = "A";
        const string TURN_RIGHT = "D";


        public MovementController()
        {
        }

        public void MonitorForMovement()
        {
            while (true)
            {
                HandleNavigation(
                    MovementModel.Instance.IsTurningLeft,
                    MovementModel.Instance.IsTurningRight,
                    MovementModel.Instance.IsDrivingForward,
                    MovementModel.Instance.IsDrivingBackward
                );
            }
        }

        private void HandleNavigation(bool isTurningLeft, bool isTurningRight, bool isDrivingForward, bool isDrivingBackward)
        {
            string ipAddress = NetworkModel.Instance.IpAddress;
            string port = NetworkModel.Instance.Port;
            IClient networkClient = NetworkModel.Instance.NetworkClient;

            if (isDrivingForward && isTurningLeft)
            {
                networkClient.Send(ipAddress, port, DRIVE_FORWARD_LEFT);
            }
            else if (isDrivingForward && isTurningRight)
            {
                networkClient.Send(ipAddress, port, DRIVE_FORWARD_RIGHT);
            }
            else if (isDrivingBackward && isTurningLeft)
            {
                networkClient.Send(ipAddress, port, DRIVE_BACKWARD_LEFT);
            }
            else if (isDrivingBackward && isTurningRight)
            {
                networkClient.Send(ipAddress, port, DRIVE_BACKWARD_RIGHT);
            }
            else if (isDrivingForward)
            {
                networkClient.Send(ipAddress, port, DRIVE_FORWARD);
            }
            else if (isDrivingBackward)
            {
                networkClient.Send(ipAddress, port, DRIVE_BACKWARD);
            }
            else if (isTurningLeft)
            {
                networkClient.Send(ipAddress, port, TURN_LEFT);
            }
            else if (isTurningRight)
            {
                networkClient.Send(ipAddress, port, TURN_RIGHT);
            }
        }
    }
}

File: RPi/Controllers/MovementController.py


class MovementController:
	DRIVE_FORWARD = 'W'
	DRIVE_FORWARD_LEFT = 'q'
	DRIVE_FORWARD_RIGHT = 'e'
	DRIVE_BACKWARD = 'S'
	DRIVE_BACKWARD_LEFT = 'z'
	DRIVE_BACKWARD_RIGHT = 'x'
	TURN_LEFT = 'A'
	TURN_RIGHT = 'D'

	def __init__(self, motor):
		self.motor = motor
		self.allowedMovements = [
			self.DRIVE_FORWARD, self.DRIVE_FORWARD_LEFT,
			self.DRIVE_FORWARD_RIGHT, self.DRIVE_BACKWARD, 
			self.DRIVE_BACKWARD_LEFT, self.DRIVE_BACKWARD_RIGHT,
			self.TURN_LEFT, self.TURN_RIGHT ]

	def onReceiveMovement(self, movement):
		if movement in self.allowedMovements:
			if movement == self.DRIVE_FORWARD:
				self.motor.driveForward()
			elif movement == self.DRIVE_FORWARD_LEFT:
				self.motor.driveForward()
				self.motor.turnLeft()
			elif movement == self.DRIVE_FORWARD_RIGHT:
				self.motor.driveForward()
				self.motor.turnRight()
			elif movement == self.DRIVE_BACKWARD:
				self.motor.driveBackward()
			elif movement == self.DRIVE_BACKWARD_LEFT:
				self.motor.driveBackward()
				self.motor.turnLeft()
			elif movement == self.DRIVE_BACKWARD_RIGHT:
				self.motor.driveBackward()
				self.motor.turnRight()
			elif movement == self.TURN_LEFT:
				self.motor.turnLeft()
			else:
				self.motor.turnRight()
